client/checkout/classic/upe.js

```diff
@@ -806,6 +806,13 @@ jQuery( function ( $ ) {
 
 	// Handle the Pay for Order form if WooCommerce Payments is chosen.
 	$( '#order_review' ).on( 'submit', () => {
+		// Skip handling legacy cards as UPE payment methods.
+		if (
+			'woocommerce_payments' ===
+			$( "#order_review input:checked[name='payment_method']" ).val()
+		) {
+			return;
+		}
```

Comment:
This whole handler for the Pay for Order page is still tailored to the old UPE<sup>TM</sup>. This initial check for CC is needed and does allow CCs to skip the UPE payment flow. We have this `isWCPayChosen()` that checks if `card` is selected, so maybe we could use that here: 

```suggestion
		if ( isWCPayChosen() ) {
			return;
		}
```

Additionally, the [`isWCPayChosen()` condition on line 819](https://github.com/Automattic/woocommerce-payments/blob/7587606cecf15107080a19f2707dcce955e66628/client/checkout/classic/upe.js#L819) should be removed or possibly replaced with a check that ensures a UPE payment method is chosen. In the old UPE, there was only a single payment gateway for both CC and UPE methods, so this check was used to prevent this code from running for non-WCPay payment methods. UPE payments are currently failing on the Pay for Order page and that `isWCPayChosen()` check is the first reason. It prevents UPE payments from ever reaching the `handleUPEOrderPay()` function.  The problems don't stop there, however. Once `handleUPEOrderPay()` is reached, we get an error when updating the payment intent due to the Stripe payment method being `undefined`. A request like this is made - `http://localhost:8082/?wc-ajax=wcpay_update_payment_intent_undefined` when it should have the payment method appended, like - `http://localhost:8082/?wc-ajax=wcpay_update_payment_intent_bancontact`. The payment method is `undefined` because we're trying to grab the value of a [dom element that no longer exists here](https://github.com/Automattic/woocommerce-payments/blob/7587606cecf15107080a19f2707dcce955e66628/client/checkout/classic/upe.js#L522). We can simply replace that with `paymentMethodType` and UPE payments will work again. We're also trying to pull in `country` from a bygone dom element which is leading to the country being `undefined` as well. I'm not sure what the implications of this are or how we can pull in country from the Pay for Order page.

Reply:
Thanks! 

> should be removed or possibly replaced with a check that ensures a UPE payment method is chosen
I'd say just removed - no need to add a check right after we performed an opposite check a few lines above. Removed `isWCPayChosen()` from this if statement. 👍 

I've also updated `form#add_payment_method` handling for adding payment methods and am using `isWCPayChosen()` there as well. May I kindly ask you to test adding CC payment methods in My account page as well?

> We're also trying to pull in country from a bygone dom element which is leading to the country being undefined as well. I'm not sure what the implications of this are or how we can pull in country from the Pay for Order page.

With the old UPE on `develop` we send empty countries for this field as well - only CC is provided, so I guess there shouldn't be any implications of this.

Reply:
>I'd say just removed - no need to add a check right after we performed an opposite check a few lines above. Removed isWCPayChosen() from this if statement. 👍

It does seem counterintuitive at first glance. The `isWCPayChosen()` condition that was removed was old UPEs way of preventing that block from executing for any payment method what was not part of WooCommerce Payments(CC and all UPE methods). Without a defensive check here, ensuring that a UPE payment method is selected, any selected payment gateway(e.g., Stripe, Cash on Delivery, PayPal) will be fed into `handleUPEOrderPay()` and will error. So we need something like a `isUPEMethodChosen()` here. As a quick test, I enabled Cash on Delivery and attempted to pay for an order. The UI was blocked indefinitely and an error appears in the console: 

```
Uncaught (in promise) TypeError: upeComponents is undefined
    checkUPEForm upe.js:458
    handleUPEOrderPay upe.js:505
    <anonymous> upe.js:814
    jQuery 9
    init checkout.min.js:1
    <anonymous> checkout.min.js:1
    jQuery 13
upe.js:458:21
```

>I've also updated form#add_payment_method handling for adding payment methods and am using isWCPayChosen() there as well. May I kindly ask you to test adding CC payment methods in My account page as well? (I've also updated the PR description).

Thanks for swapping that out as well. Adding CC methods from the My Account page is still working well.

>With the old UPE on develop we send empty countries for this field as well - only CC provides a value, so I guess there shouldn't be any implications of this. This area should be further investigated and taken care of separately - I'll note it down and make sure we iterate here after split UPE is merged.

That's great to know. Thanks for checking on this!

Reply:
Huge thanks for this vital and valid feedback. I didn't consider groups other than UPE and CC, which we work with daily.  I didn't think about any other payment method types, and your comment is super important. Thanks for it. Now coming back to the code: 

Given that there is more than one group of payment methods that the UPE implementation shouldn't handle, I think it is better not to check
```js
if (isSelectedPaymentMethodOfCCType() || isSelectedPaymentMethodOfBankTransferOrCashDepositEtc())
```


but to 
```js
if(isSelectedPaymentMethodNotAUPE())
```

so I just used `getSelectedGatewayPaymentMethod()` function that handles only UPE and returns `null` in any other way. Thanks to this method, we can exclude all the other payment methods. The commit is there, may I ask you to test this out and let me know, what do you think? Thanks.